vercel/src/lib/AsciiPreview.svelte

```diff
@@ -0,0 +1,65 @@
+<script lang="ts">
+  import type { LanguageColors } from '../types';
+
+  export let name: string;
+  export let ansi: string[];
+  export let hex: string[] | null = null;
+  export let ascii: string = '';
+
+  let dark = true;
+  let trueColor = hex != null;
+
+  $: html = ascii
+    .split('\n')
+    .map((line) => {
+      // TODO Clean up, this is hard to read
+      let spanCount = 0;
+      const htmlLine = line.replace(/\{(\d+)\}/g, (_match, index) => {
+        const i = Number.parseInt(index, 10);
+        const spanText = trueColor
+          ? `<span style="color: ${hex[i]}">`
+          : `<span class="text-ansi-${ansi[i]}">`;
```

Comment:
We should add the terminal default foreground logic: `if white (or #FFFFFF) && !Dark --> Black`